UnityProject/Assets/Scripts/Objects/VisibleBehaviour.cs

```diff
@@ -93,7 +106,19 @@ private void UpdateState(bool _aliveState)
 
 		for (int i = 0; i < renderers.Length; i++)
 		{
-			renderers[i].enabled = _aliveState;
+			// Cast and check cast
+			SpriteRenderer sr = renderers[i] as SpriteRenderer;
+			if (sr != null)
+			{
+				if (CanBeDisabled(sr))
+				{
+					sr.enabled = _aliveState;
+				}
+			}
+			else
+			{
+				renderers[i].enabled = _aliveState;

```

Comment:
This is weird because you are saying that if renderers does not contain a renderer then disable the renderer that according to this condition should not contain a renderer. Very confusing. You don't need this at all. Just remove the whole condition and check if:
```
if(CanBeDisabled(renderers[i]){
renderers[i].enabled = _aliveState;
}
```